### PR TITLE
feat: Implement method to get plugin config properties

### PR DESCRIPTION
**Pull request by prateek3255**
## Summary of change

- Add core config method to get plugin config properties

## Related issues

N/A

## Test Plan

N/A

## Documentation changes
N/A

## Checklist for important updates
- [x] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [x] Changes to the version if needed
   - In `build.gradle`
- [x] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
N/A

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+## [5.0.8] - 2024-02-08
+
+- Adds implementation for a new method `getConfigFieldsJson` to fetch the plugin config fields.
```
Comment (sattvikc) [inline CHANGELOG.md:12]: no need of version changes now, create base branch for mfa dashboard

File: build.gradle [hunk 2]
```diff
@@ -2,7 +2,7 @@ plugins {
     id 'java-library'
 }
 
-version = "5.0.7"
+version = "5.0.8"
```
Comment (sattvikc) [inline build.gradle:5]: not required

File: src/test/java/io/supertokens/storage/postgresql/test/PostgresSQLConfigTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            assertTrue(config.get("isDifferentAcrossTenants").getAsJsonPrimitive().isBoolean());
+            assertTrue(config.get("type").getAsJsonPrimitive().isString());
+            String type = config.get("type").getAsString();
+            assertTrue(type.equals("number") || type.equals("boolean") || type.equals("string"));
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
```
Comment (sattvikc) [inline src/test/java/io/supertokens/storage/postgresql/test/PostgresSQLConfigTest.java:72]: add a test to compare config descriptions from annotations and `config.yaml` and `devConfig.yaml`

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java [hunk 4]
```diff
@@ -126,6 +148,45 @@ public static Set<String> getValidFields() {
...
+
+                String type = "string";
+
+                Class<?> fieldType = field.getType();
+
+                if (fieldType == java.lang.String.class) {
+                    type = "string";
+                } else if (fieldType == boolean.class) {
+                    type = "boolean";
+                } else if (fieldType == int.class || fieldType == long.class || fieldType == double.class) {
```
Comment (sattvikc) [inline src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java:176]: you may want to assert for any other type (make it future safe). Also add tests that all JsonConfig have description.

**Review by sattvikc — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (prateek3255) [reply CHANGELOG.md]: Fixed

File: build.gradle (hunk 2 above)
Comment (prateek3255) [reply build.gradle]: Fixed

File: src/test/java/io/supertokens/storage/postgresql/test/PostgresSQLConfigTest.java (hunk 3 above)
Comment (prateek3255) [reply src/test/java/io/supertokens/storage/postgresql/test/PostgresSQLConfigTest.java:85]: Added the test similar to core

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java (hunk 4 above)
Comment (prateek3255) [reply src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java]: Done

**Review by sattvikc — APPROVED** [review]

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java (hunk 4 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java]: not correctly implemented

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java [hunk 5]
```diff
@@ -134,6 +159,43 @@ public static Set<String> getValidFields() {
...
+
+                Class<?> fieldType = field.getType();
+
+                if (fieldType == java.lang.String.class || fieldType == char.class) {
+                    type = "string";
+                } else if (fieldType == boolean.class) {
+                    type = "boolean";
+                } else if (fieldType == byte.class || fieldType == short.class || fieldType == int.class || fieldType == long.class || fieldType == float.class || fieldType == double.class) {
+                    type = "number";
+                }
```
Comment (sattvikc) [inline src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java:188]: this is incorrect

**Review by sattvikc — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java (hunk 4 above)
Comment (prateek3255) [reply src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java]: Fixed

**Review by sattvikc — APPROVED** [review]

**Pull request merged**